File: apple/Sources/ReactNativeWatchOS/Modifiers/MenuOrder.swift
import SwiftUI

/// SwiftUI `.menuOrder(_:)`. Controls the presentation order of items inside a
/// menu — e.g. whether the first declared item sits closest to the trigger.
///
/// Available on watchOS 9+, so no availability gate is needed.
enum RNWMenuOrderModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("menuOrder") { view, params, _ in
            // `.priority` is unavailable on watchOS; only `.fixed` and
            // `.automatic` exist here. "priority" falls back to `.automatic`.
            switch params.string("order") {
            case "fixed":    return AnyView(view.menuOrder(.fixed))
            default:         return AnyView(view.menuOrder(.automatic))
            }
        }
    }
}

File: src/modifiers/menuOrder.ts
import { createModifier } from './createModifier';

/// The order menu items are presented in.
/// - `'priority'` keeps the declared order, anchored nearest the menu trigger
/// - `'fixed'` keeps the declared order regardless of placement
export type MenuOrderValue = 'automatic' | 'fixed' | 'priority';

/// SwiftUI `.menuOrder(_:)`. Controls the presentation order of items inside a
/// menu (e.g. whether the first item sits closest to the user's finger).
export function menuOrder(order: MenuOrderValue = 'automatic') {
  return createModifier('menuOrder', { order });
}
